projects/hathor-wallet-mobile/0002-nano-contract-integration.md

```diff
@@ -0,0 +1,476 @@
+	- Differentiate item *Action List* item types *deposit* and *withdrawal* using distict visual indicators.
+
+- **Error Management**:
+	- In case of processing error, display an error feedback.
+
+>[!NOTE]
+>The *Caller* represents the wallet address calling the Blueprint method.
+
+>[!TIP]
+>Use the following value examples:
+>- *Blueprint ID*
+>	- eg: `494d0ac59a6918b771b122395206fef3f349f84f20dc430188a319d4ead24a3b`
+>- *Blueprint Method*
+>	- eg: `initiale`, `swap`, `bet`
+>- *Caller*
+>	- eg: `WRqBSi5jc74P8tRo7CmgiGByA1eTzfMh7B`
+>- *Nano Contract ID*
+>	- eg: `000001e43dda7348c53c6252d98799c9e2a0b4ea3b5f10b1182578d643c9641d`
+>- *Action List* item:
+>	- *Type*
+>		- eg: `deposit`, `withdrawal`
+>	- *Token*
+>		- eg: `000001a176aa72b84f3486b3fcea327b023a70d1b6461c1b6a9073785b5775d8`
+>	- *Data*
+>		- *Amount*
+>			- eg: `1.23`
+>		- *Address*
+>			- eg: `WZhKusv57pvzotZrf4s7yt7P7PXEqyFTHk`
+>- *Argument List* item:
+>	- *Type*
+>		- eg: `byte`, `int`, `string`, `boolean`
+>	- *Value*
+>		- eg: `000002d068a46c1c0a6ae84ca3b3c1984d517c8a3783c2ceb024cd5355595851` for byte
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Load Nano Contract History
+
+##### When execute
+- It should happen everytime a new *Nano Contract Transaction* containing a caller from user's wallet is processed on `sagas/wallet.js@handleTx`.
+- It should happen when the *Nano Contract Transaction List* component is mounted, for the selected *Nano Contract*, for the first time and there isn't a history registered.
+
+##### Where to execute
+- On `sagas/nanoContract.js@fetchNanoContractHistory`:
+	- It should use `ncApi.getNanoContracHistory` from wallet-lib.
+
+##### Where to store
+- On redux state `nanoContractsHistory`:
+	- It must be a map of `<TxHash, NcTxHistory>`.
+	- The `NcTxHistory` must contain a set of balance per token.
+	- The `NcTxHistory` must contain relevante information such as:
+	 	- *Transaction ID*
+		- *Nano Contract ID*
+		- *Blueprint Method*
+		- *Argument List*
+		- *Timestamp*
+
+## Register Nano Contract
+
+##### When execute
+- It should happen when a *Nano Contract* is registered using the *Nano Contract Registration* flow.
+
+##### Where to execute
+- On `sagas/nanoContract.js@registerNanoContract`.
+
+##### Where to store
+- On redux state `registeredNanoContracts`.
+- On AsyncStore using a key `nanoContract:registered` for persistence, once the lib doesn't have an interface on the `store`.
+
+## Wallet Connect
+
+### RPC Protocol
+
+#### `htr_createNcTx`
+[htr_createnctx]: #htr_createnctx
+
+The `htr_createNcTx` method creates a new *Nano Contract* transaction based on parameters. It will select the utxos automatically, mine and send the transaction.
+
+##### Parameters
+
+- `blueprintId`: unique identification of the *Blueprint*
+- `method`: *Blueprint*'s method name
+- `caller`: wallet address representing method's *Caller*
```

Comment:
Can you give more context here on the nano contracts specific parameters or point to the documentation?

Reply:
See the following [signature](https://github.com/HathorNetwork/hathor-wallet-lib/blob/50dfe66e720b481619e77558ddfc83e7096bae27/src/new/wallet.js#L2619):
```ts
async createAndSendNanoContractTransaction(blueprintId, method, address, data, options = {})
```